File: activity/src/main/java/dev/sayaya/handbook/client/domain/WorkspaceBuilder.java
package dev.sayaya.handbook.client.domain;

import lombok.Setter;
import lombok.experimental.Accessors;

@Setter
@Accessors(fluent = true)
public class WorkspaceBuilder {
    private String id;
    private String name;
    public Workspace build() {
        return Workspace.of(id, name);
    }
}

File: persist-workspace/src/main/kotlin/dev/sayaya/handbook/usecase/WorkspaceBuilder.kt
package dev.sayaya.handbook.usecase

import dev.sayaya.handbook.domain.Workspace
import java.util.*

data class WorkspaceBuilder (
    var id: UUID?=null,
    var name: String,
    var description: String?=null,
){
    fun build() = Workspace (
        id ?: UUID.randomUUID(),
        name,
        description
    )
}